pkg/upgradeservice/preflight/preflight.go

```diff
@@ -142,31 +147,61 @@ func Run(app *apptypes.App, archiveDir string, sequence int64, registrySettings
 	return nil
 }
 
-func setPreflightResults(results *types.PreflightResults) error {
-	preflightData, err := getPreflightData()
+func setPreflightResults(appSlug string, results *types.PreflightResults) error {
+	preflightData, err := GetPreflightData()
 	if err != nil {
 		return errors.Wrap(err, "failed to get preflight data")
 	}
-	preflightData.Results = results
+	resultsBytes, err := json.Marshal(results)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal preflight results")
+	}
+	createdAt := time.Now()
+	preflightData.Result = &types.PreflightResult{
+		Result:                     string(resultsBytes),
+		CreatedAt:                  &createdAt,
+		AppSlug:                    appSlug,
+		ClusterSlug:                "this-cluster",
+		Skipped:                    false,
+		HasFailingStrictPreflights: hasFailingStrictPreflights(results),
+	}
+	// clear the progress once the results are set
+	preflightData.Progress = ""
 	if err := setPreflightData(preflightData); err != nil {
 		return errors.Wrap(err, "failed to set preflight results")
 	}
 	return nil
 }
 
+func hasFailingStrictPreflights(results *types.PreflightResults) bool {
+	// convert to troubleshoot type so we can use the existing function
+	uploadResults := &troubleshootpreflight.UploadPreflightResults{}
+	uploadResults.Results = results.Results
+	for _, e := range results.Errors {
+		uploadResults.Errors = append(uploadResults.Errors, &troubleshootpreflight.UploadPreflightError{
+			Error: e.Error,
+		})
+	}
+	return troubleshootpreflight.HasStrictAnalyzersFailed(uploadResults)
+}
```

Comment:
This felt weird to write, but there are two types that are nearly identical [`PreflightResults`](https://github.com/replicatedhq/kots/blob/2002ad802ee9cabb2595d5b5297f4425219956b8/pkg/preflight/types/types.go#L23-L26) and [`UploadPreflightResults`](https://github.com/replicatedhq/troubleshoot/blob/a9c5baca1464d9e8d87d2817aa540985114530b4/pkg/preflight/types.go#L18-L21), the latter of which is from troubleshoot. I wanted to keep using the same check for failing strict preflights, but if there's a better way to do this, I'm all ears.

Reply:
it's fine for now.